### ISSUE TITLE
Jensen–Shannon divergence calculation

### ISSUE THREAD

**Issue by junjiah** [evansenter/diverge#1]
according to Wikipedia, Jensen–Shanno divergence should be calculated as:

```
js(p,q) = 0.5*kl(p,m)+0.5*kl(q,m), where m = 0.5(p+q)
```

rather than 

```
js(p,q) = 0.5*kl(p,q)+0.5*kl(q,p)
```

in `diverge.rb`

**Comment by evansenter** [comment]
I believe Wikipedia is incorrect / they are analogous. Take a look at the following references:

Jeffreys, H. (1946). An Invariant Form for the Prior Probability in Estimation Problems. Proceedings of the Royal Society of London. Series A. Mathematical and Physical Sciences, 186(1007), 453–461. doi:10.1098/rspa.1946.0056
Lin, J. (1991). Divergence measures based on the Shannon entropy. Information Theory, IEEE Transactions on, 37(1), 145–151. doi:10.1109/18.61115

In particular, equation 2.2 of Lin J., 1991.

**Comment by junjiah** [comment]
That is  _J-divergence_ (Jeffreys, 1946; Kullback and Leibler, 1961), while _Jensen difference divergence_ or _Jensen–Shanno divergence_ is proposed by Sibson (1969).

Check the first two lines [here](http://www.mtm.ufsc.br/~taneja/book/node21.html).

**Comment by evansenter** [comment]
Take a look at https://github.com/evansenter/diverge/commit/dc85869a7cd2ba54cfb177a07ff3f1f003e23026

Should fix the issue, if you agree I'll re-push the gem.

**Comment by evansenter** [comment]
Fixed in dc85869, updated revision to reflect bugfix and republished 1.5.2 version on Rubygems.
